views.py

```diff
@@ -53,45 +52,54 @@ def download(request, path, *args, **kwargs):
         else:
             raise KeyError('settings must have IRODS_GLOBAL_SESSION set if there is no environment object')
 
+    is_bag_download = False
     if split_path_strs[0] == 'bags':
         res_id = os.path.splitext(split_path_strs[1])[0]
+        is_bag_download = True
     else:
         res_id = split_path_strs[0]
     res, authorized, _ = authorize(request, res_id, needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE,
                                    raises_exception=False)
     if not authorized:
```

Comment:
In case of REST API call to this function, we should be raising ```PermissionDenied``` if authorization fails.